packages/runtime/datastore/src/dataStoreRuntime.ts

```diff
@@ -377,38 +391,52 @@ IFluidDataStoreChannel, IFluidDataStoreRuntime, IFluidHandleContext {
         if (this.isAttached) {
             this.attachChannel(channel);
             return;
-        } else {
-            this.bind(channel.handle);
+        }
 
-            // If our data store is local then add the channel to the queue
-            if (!this.localChannelContextQueue.has(channel.id)) {
-                this.localChannelContextQueue.set(channel.id, this.contexts.get(channel.id) as LocalChannelContextBase);
-            }
+        /**
+         * If this channel has already been bound, do nothing. This can happen during attachGraph() when a channel's
+         * graph is attached. It calls bindToContext on the shared object which will end up back here.
```

Comment:
Do we have test coverage for all cases like that?

Reply:
Yes, this is very common scenario and will happen for all DDSs that are created in detached container.